### PR TITLE
Improve tolerance of paddles faults

### PR DESCRIPTION
**Pull request by zmc**

### CODE REVIEW

**Review by dmick — APPROVED** [review]
I guess.  Endless loops always worry me, but...desperate times.

**Comment by zmc** [comment]
> I guess. Endless loops always worry me, but...desperate times.

It's a daemon!

File: teuthology/lock/query.py [hunk 1]
```diff
@@ -51,7 +51,7 @@ def is_vm(name=None, status=None):
     return status.get('is_vm', False)
 
 
-def list_locks(keyed_by_name=False, **kwargs):
+def list_locks(keyed_by_name=False, tries=10, **kwargs):
```
Comment (kshtsk) [inline teuthology/lock/query.py:54]: @zmc ...so, please, tell me why there is the _tries_ parameter if it is unused? Also giving an endless loop makes me frustrating when I run `teuthology_suite` to without lock server configured, and given that timeout is zero... I am experiencing trouble to find point where the error occurred.

Comment (zmc) [reply teuthology/lock/query.py:54]: The `tries` parameter is not unused.

I don't understand the issue you're describing, but this is really not the place to report or discuss it. A tracker ticket perhaps?

Comment (kshtsk) [reply teuthology/lock/query.py:54]: @zmc tracker ticket looks overkill for me for sharing some thoughts, full of formalism and bureaucracy :-)

Comment (kshtsk) [reply teuthology/lock/query.py:54]: > The `tries` parameter is not unused.

Look carefully, there is no usage of `tries` value in `list_locks`.

File: teuthology/lock/query.py [hunk 2]
```diff
@@ -63,7 +63,11 @@ def list_locks(keyed_by_name=False, **kwargs):
             kwargs['machine_type'] = kwargs['machine_type'].replace(',','|')
         uri += '?' + urlencode(kwargs)
     with safe_while(
-            sleep=1, increment=0.5, action='list_locks') as proceed:
+            sleep=1,
+            increment=0.5,
+            tries=-1,
```
Comment (kshtsk) [inline teuthology/lock/query.py:68]: It's hardcoded here to be infinite.

**Pull request merged**